crates/bevy_core_pipeline/src/taa/taa.wgsl

```diff
@@ -65,43 +69,69 @@ fn clip_towards_aabb_center(history_color: vec3<f32>, current_color: vec3<f32>,
 }
 
 fn sample_history(u: f32, v: f32) -> vec3<f32> {
-    return textureSample(history, linear_sampler, vec2(u, v)).rgb;
+    return textureSampleLevel(history, linear_sampler, vec2(u, v), 0.0).rgb;
 }
 
 fn sample_view_target(uv: vec2<f32>) -> vec3<f32> {
-    var sample = textureSample(view_target, nearest_sampler, uv).rgb;
+    var sample = textureSampleLevel(view_target, nearest_sampler, uv, 0.0).rgb;
 #ifdef TONEMAP
     sample = tonemap(sample);
 #endif
     return RGB_to_YCoCg(sample);
 }
 
+fn disocclusion_checks(uv: vec2<f32>, history_uv: vec2<f32>, current_depth_nonlinear: f32) -> bool {
+    // Check if history is now off-screen
+    if any(history_uv < 0.0) || any(history_uv > 1.0) {
+       return false;
+    }
+
+    // Compare current and previous depth
+    let current_depth = 1.0 / current_depth_nonlinear;
+    let last_frame_depth = 1.0 / textureSampleLevel(last_frame_depth, nearest_sampler, history_uv, 0.0);
+    if abs(current_depth - last_frame_depth) > 0.05 {
```

Comment:
@JMS55 I don't think we should use the linear depth here. It will cause TAA to act very differently depending on how close the camera is to things. Thing far in the distance will now be much more likely to trigger disocclusion. I think we should probably use the pixel radius, or non linear depth somehow. I haven't looked yet at what is typically used for TAA.

I noticed in this version that there's a lot more aliasing under motion than before on the edges of the boxes as well as on the noisy plane.

Reply:
Typically you check for disocclusion by checking if the difference in depth is greater than ~10%. I thought the depth of the texture was _already_ non-linear. We should probably be using world/view space depth. I always forget what the depth texture is stored as though... (we could really use some docs about the prepass formats).

There's also a plane distance check that has less positive disocclusions, but I've left that for future work.

> I noticed in this version that there's a lot more aliasing under motion than before on the edges of the boxes as well as on the noisy plane.

Can you try plugging in `closest_depth` instead of `d_mm` into the disocclusion checks and see if that fixes it?

Reply:
Sounds like we should just use linear depth (so that 0.1 and 0.2 are the same distance apart as 0.9 and 1.0, right?) then.

But you said 
> I don't think we should use the linear depth here. It will cause TAA to act very differently depending on how close the camera is to things. 
which I don't understand why. Did you mean non-linear depth?

Reply:
Also, not sure how well we'll actually be able to handle the noisy pixelated plane, that's kinda a torture test. I'm more concerned about regular geometry and materials.

Reply:
I don't think we should use linear depth. Yes 0.1 and 0.2 are the same distance apart as 0.9 and 1.0, but depending how close the camera is to something it will dramatically change the distances involved. Things in the far distance will often much more easily have significantly larger distance differences, while appearing close together on screen. 

One example of how to think about this it to think about a normal sized scene, say the indoor part of bistro. If you scale the whole scene up 10x and make the camera move 10x faster, every thing will look and feel the same, but the distances will be a lot further and TAA will totally break. On the other hand if you scale everything down to miniature size, make the camera move slower, it will again look the same, but TAA will not disocclude nearly as much as at normal scale.

I think your initial comment about "checking if the difference in depth is greater than ~10%" is probably closer to what we want and would require either not using linear depth or taking into account the pixel radius.